common/src/main/java/io/netty/util/internal/CleanerJava6.java

```diff
@@ -29,39 +29,29 @@
  *
  * For more details see <a href="https://github.com/netty/netty/issues/2604">#2604</a>.
  */
-final class Cleaner0 {
+final class CleanerJava6 implements Cleaner {
     private static final long CLEANER_FIELD_OFFSET;
     private static final Method CLEAN_METHOD;
-    private static final boolean CLEANER_IS_RUNNABLE;
 
-    private static final InternalLogger logger = InternalLoggerFactory.getInstance(Cleaner0.class);
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(CleanerJava6.class);
 
     static {
         ByteBuffer direct = ByteBuffer.allocateDirect(1);
         Field cleanerField;
         long fieldOffset = -1;
         Method clean = null;
-        boolean cleanerIsRunnable = false;
         Throwable error = null;
         if (PlatformDependent0.hasUnsafe()) {
             try {
                 cleanerField = direct.getClass().getDeclaredField("cleaner");
                 fieldOffset = PlatformDependent0.objectFieldOffset(cleanerField);
                 Object cleaner = PlatformDependent0.getObject(direct, fieldOffset);
-                try {
-                    // Cleaner implements Runnable from JDK9 onwards.
-                    Runnable runnable = (Runnable) cleaner;
-                    runnable.run();
-                    cleanerIsRunnable = true;
-                } catch (ClassCastException ignored) {
-                    clean = cleaner.getClass().getDeclaredMethod("clean");
-                    clean.invoke(cleaner);
-                }
+                clean = cleaner.getClass().getDeclaredMethod("clean");
+                clean.invoke(cleaner);
             } catch (Throwable t) {
                 // We don't have ByteBuffer.cleaner().
                 fieldOffset = -1;
                 clean = null;
-                cleanerIsRunnable = false;
                 error = t;
             }
         }
```

Comment:
the conditional below checks if `error == null` and then says "cleaner is available". However this may not be the case if we have no unsafe. We should adjust the logic below or set the `error = <exception indicating unsafe isn't available>`.